src/clue.js

```diff
@@ -1,14 +1,208 @@
+        image: "https://www.radiotimes.com/uploads/images/Original/111967.jpg",
+        color: "red",
+    },
+    {
+        //mrsPeacock,
+        firstName: "Eleanor",
+        lastName: "Peacock",
+        occupation: "Socialité",
+        age: 36,
+        description: "She is from a wealthy family and uses her status and money to earn popularity",
+        image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+        color: "blue",
+    },
+    {
+        //mrMustard,
+        firstName: "Jack",
+        lastName: "Mustard",
+        occupation: "Retired Football player",
+        age: 62,
+        description: "He is a former football player who tries to get by on his former glory",
+        image: "https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg",
+        color: "yellow",
+    }
+];
 
 // Rooms Collection
-const roomsArray = [];
+const roomsArray = [
+    {
+        name: "Dining Room"
+    },
+    {
+        name: "Conservatory"
+    },
+    {
+        name: "Kitchen"
+    },
+    {
+        name: "Study"
+    },
+    {
+        name: "Library"
+    },
+    {
+        name: "Billiard Room"
+    },
+    {
+        name: "Lounge"
+    },
+    {
+        name: "Ballroom"
+    },
+    {
+        name: "Hall"
+    },
+    {
+        name: "Spa"
+    },
+    {
+        name: "Living Room"
+    },
+    {
+        name: "Observatory"
+    },
+    {
+        name: "Theater"
+    },
+    {
+        name: "Guest House"
+    },
+    {
+        name: "Patio"
+    }
+];
 
 // Weapons Collection
-const weaponsArray = [];
+const weaponsArray = [
+    {
+        name: "rope",
+        weight: 10,
+    },
+
+    {
+        name: "knife",
+        weight: 8
+    },
+
+    {
+        name: "candlestick",
+        weight: 2
+    },
+
+    {
+        name: "dumbbell",
+        weight: 30
+    },
+
+    {
+        name: "poison",
+        weight: 2
+    },
+
+    {
+        name: "axe",
+        weight: 15
+    },
+
+    {
+        name: "bat",
+        weight: 13
+    },
+
+    {
+        name: "trophy",
+        weight: 25
+    },
+
+    {
+        name: "pistol",
+        weight: 20
+    }
+];
 
 // ITERATION 2
 
+function selectRandom(arr) {
+    if (arr.length === 0) {
+        return undefined
+    } else {
+        var arr = arr[Math.floor(Math.random() * arr.length)];
+        return arr
+    }
+
+}
+
+const suspect = selectRandom(suspectsArray)
+const weapon = selectRandom(weaponsArray)
+const room = selectRandom(roomsArray)
+
+
+function pickMystery() {
+    let a = {}
+
+    a = {
+        suspect: suspect,
+        weapon: weapon,
+        room: room
+    }
+
+    console.log(a)
+    return a
+}
+
+
 // ITERATION 3
+
+const a = pickMystery()
+
+function revealMystery() {
+
+    const b = `${a.suspect.firstName} ${a.suspect.lastName} killed Mr. Boddy using the ${a.weapon.name} in the ${a.room.name}!`
+
+    console.log(b)
+    return b
+
+}
```

Comment:
```suggestion
function revealMystery(misteryObj) {

    const misterySolution = `${misteryObj.suspect.firstName} ${misteryObj.suspect.lastName} killed Mr. Boddy using the ${misteryObj.weapon.name} in the ${misteryObj.room.name}!`

    console.log(misterySolution)
    return misterySolution

}
```

Si compruebas el test verás que está pensado para recibir un obj como parámetro y emplearlo para resolver el misterio.  Si lo adaptas para eso ya funciona.